sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -75,7 +75,7 @@ public static HttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] per
 
             policies.Add(BufferResponsePolicy.Shared);
 
-            policies.Add(new RequestActivityPolicy(isDistributedTracingEnabled));
+            policies.Add(new RequestActivityPolicy(isDistributedTracingEnabled, ClientDiagnostics.GetResourceProviderNamespace(options.GetType().Assembly)));
```

Comment:
It would be faster, when possible, just to get this as a virtual property that could do this by default, but also return just a literal string name from child implementations of the base options class.

Reply:
I'd like to try to avoid any public API changes because distributed tracing requirement are quite fluid and I'd like to be able to remove this feature if required.